src/display/editor/drawers/signaturedraw.js

```diff
@@ -491,32 +576,55 @@ class SignatureExtractor {
     );
 
     const threshold = this.#guessThreshold(histogram);
-    const contourList = this.#findContours(
-      uint8Filtered,
-      width,
-      height,
-      threshold
+    const contourList = this.#findContours(buffer, width, height, threshold);
+
+    return this.processDrawnLines(
+      { curves: contourList, width, height },
+      pageWidth,
+      pageHeight,
+      rotation,
+      innerMargin,
+      true,
+      true
     );
-    const linesAndPoints = [];
+  }
 
+  static processDrawnLines(
+    lines,
+    pageWidth,
+    pageHeight,
+    rotation,
+    innerMargin,
+    mustSmooth,
+    areContours
+  ) {
```

Comment:
Given the number of parameters this method needs, I'd suggest using a parameter-object instead to reduce the risk of future bugs caused be accidentally messing up the order (especially since there's a couple of boolean parameters).